### ISSUE TITLE
useUnifiedTopology: true

### ISSUE THREAD

**Issue by jeyraj** [Automattic/mongoose#8169]
(node:1612) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pas s option { useUnifiedTopology: true } to the MongoClient constructor.
--

I am getting the above warning  with latest mongoose and mongodb.
I am using 
Nodejs   10.16.x
MongoDB Server  4.2
Mongoose  5.7.1

I configured mongoose properly with below options as well  
               ` mongoose.set('useUnifiedTopology', true);  
                 mongoose.set('useNewUrlParser', true);   
                  mongoose.connect()`

But still getting topology related warning.

**Comment by vkarpov15** [comment]
The below script doesn't print any warnings:

```javascript
const assert = require('assert');
const mongoose = require('mongoose');
mongoose.set('debug', true);

const GITHUB_ISSUE = `gh8169`;
const connectionString = `mongodb://localhost:27017/${ GITHUB_ISSUE }`;
const { Schema } = mongoose;

run().then(() => console.log('done')).catch(error => console.error(error.stack));

async function run() {
  mongoose.set('useUnifiedTopology', true);
  mongoose.set('useNewUrlParser', true);
  await mongoose.connect(connectionString);
} 
```

Are you sure you're calling `mongoose.set('useUnifiedTopology', true)` **before** calling [`mongoose.connect()`](https://mongoosejs.com/docs/api/mongoose.html#mongoose_Mongoose-connect)?

**Comment by jeyraj** [comment]
Yes.. I am calling mongoose.set('useUnifiedTopology', true) before mongoose.connect().

![code](https://user-images.githubusercontent.com/55078977/65577287-22e20300-df91-11e9-876f-956983a89b60.png)

**Comment by vkarpov15** [comment]
Are you sure you're on v5.7.1? Print out [the `mongoose.version` to be sure](https://mongoosejs.com/docs/api/mongoose.html#mongoose_Mongoose-version). The below script doesn't print any warnings for me:

```javascript
const assert = require('assert');
const mongoose = require('mongoose');
mongoose.set('debug', true);

const GITHUB_ISSUE = `gh8169`;
const connectionString = `mongodb://localhost:27017/${ GITHUB_ISSUE }`;
const { Schema } = mongoose;

run().then(() => console.log('done')).catch(error => console.error(error.stack));

async function run() {
  console.log('Mongoose version:', mongoose.version);
  mongoose.set('useUnifiedTopology', true);
  mongoose.set('useNewUrlParser', true);
  mongoose.connect(connectionString, { useFindAndModify: false });
  const db = mongoose.connection;
  
  db.once('open', () => console.log('connected'));
} 
```

Output:

```
$ node gh-8169.js 
Mongoose version: 5.7.1
done
connected
^C
$ 
```

**Comment by stevenventimiglia** [comment]
I want to add to this thread that it may also have to do with other dependencies.

For instance, nothing I updated or set for NodeJS, MongoDB or Mongoose were the issue - however - `connect-mongodb-session` had been updated and starting slinging the same error. The solution, in this case, was to simply rollback the version of `connect-mongodb-session` from version 2.3.0 to 2.2.0.

https://imgur.com/a/Vr867X9
